reconcile/terraform_resources.py

```diff
@@ -716,3 +721,25 @@ def run(
         cleanup_and_exit(tf, err)
 
     cleanup_and_exit(tf)
+
+
+def early_exit_desired_state(*args, **kwargs) -> dict[str, Any]:
+    gqlapi = gql.get_api()
+    namespaces = get_tf_namespaces()
+    resources = []
+    for ns_info in namespaces:
+        for spec in get_external_resource_specs(
+            ns_info, provision_provider=PROVIDER_AWS
+        ):
+            defaults = spec.resource.get("defaults")
+            if defaults:
+                resources.append(gqlapi.get_resource(defaults))
```

Comment:
Nit: This is again my opportunity to shamelessly plug resource ref loading 😊

We should consider it for a followup PR